userspace/libsinsp/cri.hpp

```diff
@@ -782,23 +767,32 @@ inline bool cri_interface<api>::parse(const libsinsp::cgroup_limits::cgroup_limi
 		container.m_cpu_quota = limits.m_cpu_quota;
 		container.m_cpu_period = limits.m_cpu_period;
 		container.m_cpuset_cpu_count = limits.m_cpuset_cpu_count;
-
-		// In some cases (e.g. openshift), the cri-o response
-		// may not have an info property, which is used to set
-		// the container user. In those cases, the container
-		// name stays at its default "<NA>" value.
 	}
 
-	g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s): after parse_containerd: repo=%s tag=%s image=%s digest=%s",
+	g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s): after container parsing: repo=%s tag=%s image=%s digest=%s",
 			container.m_id.c_str(), container.m_imagerepo.c_str(), container.m_imagetag.c_str(),
 			container.m_image.c_str(), container.m_imagedigest.c_str());
 
+	// Enabled by default for Falco consumer
 	if(s_cri_extra_queries)
 	{
-		if(!container.m_container_ip)
-		{
-			get_container_ip(container.m_id, container.m_container_ip, container.m_pod_cniresult);
-		}
+		/*
+		* The recent refactor makes full use of PodSandboxStatusResponse, removing the need to access pod sandbox containers
+		* in k8s filterchecks. Now, we also store the pod sandbox labels in the container.
+		* While this might seem redundant in cases where multiple containers exist in a pod, considering that the concurrent
+		* number of containers on a node is typically capped at 100-300 and many pods contain only 1-3 containers,
+		* it  doesn't add significant overhead. Moreover, these extra lookups have always been performed for container ips in the past
+		* and therefore are no new additions.
+		*/
+		typename api::PodSandboxStatusResponse pod_sandbox_status_resp;
+		status = get_pod_sandbox_status_resp(container.m_pod_sandbox_id, pod_sandbox_status_resp);
+		const auto &resp_pod_sandbox_container = pod_sandbox_status_resp.status();
+		const auto &resp_pod_sandbox_container_info = pod_sandbox_status_resp.info();
+		const auto root_pod_sandbox = get_info_jvalue(resp_pod_sandbox_container_info);
+		// Add pod response network and labels to original container
+		parse_cri_pod_sandbox_network(resp_pod_sandbox_container, root_pod_sandbox, container);
```

Comment:
I mentioned before that we may have done the network enrichment twice before and now only one time in a transparent and clean way.

Reply:
@gnosek highlighting again that I am not introducing extra CRI API calls. They were already there before. 

In summary, the design is that for each container that is not a pod sandbox container you get a `ContainerStatsResponse` once and also a `PodSandboxStatusResponse` once and fully make use of the fact that you have the `PodSandboxStatusResponse`.

If we want to complicate things and cache `PodSandboxStatusResponse` info in pod sandbox containers so that the call is only done once for each pod is a different discussion. I would not recommend complicating things for now, but we could think about it in the future.